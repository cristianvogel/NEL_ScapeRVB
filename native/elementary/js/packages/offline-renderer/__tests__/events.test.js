import OfflineRenderer from '../index';
import { el } from '@elemaudio/core';


test('event propagation', async function() {
  let core = new OfflineRenderer();

  await core.initialize({
    numInputChannels: 0,
    numOutputChannels: 1,
    blockSize: 512,
  });

  // Event handling
  let callback = jest.fn();
  core.on('meter', callback);

  // Graph
  core.render(el.meter(0));

  // Processing
  let inps = [];
  let outs = [new Float32Array(512 * 4)];

  core.process(inps, outs);

  // We just pushed four blocks of data through, we would expect the meter
  // node to fire on each block, thus we should see 4 calls to the meter
  // callback
  expect(callback.mock.calls).toHaveLength(4);
  expect(callback.mock.calls.map(x => x[0])).toMatchSnapshot();

  callback.mockClear();

  // New graph
  core.render(el.meter(1));

  // Processing
  core.process(inps, outs);

  // We just pushed four more blocks of data, we would expect the same
  // as above but this time the meter should be reporting the new value 1
  expect(callback.mock.calls).toHaveLength(4);
  expect(callback.mock.calls.map(x => x[0])).toMatchSnapshot();
});
