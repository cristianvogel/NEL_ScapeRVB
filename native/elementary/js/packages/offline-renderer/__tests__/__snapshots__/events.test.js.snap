// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`event propagation 1`] = `
Array [
  Object {
    "max": 0,
    "min": 0,
    "source": undefined,
  },
  Object {
    "max": 0,
    "min": 0,
    "source": undefined,
  },
  Object {
    "max": 0,
    "min": 0,
    "source": undefined,
  },
  Object {
    "max": 0,
    "min": 0,
    "source": undefined,
  },
]
`;

exports[`event propagation 2`] = `
Array [
  Object {
    "max": 1,
    "min": 1,
    "source": undefined,
  },
  Object {
    "max": 1,
    "min": 1,
    "source": undefined,
  },
  Object {
    "max": 1,
    "min": 1,
    "source": undefined,
  },
  Object {
    "max": 1,
    "min": 1,
    "source": undefined,
  },
]
`;
